### PR TITLE
Basic basepath support

### PR DESCRIPTION
**Pull request by omgovich**
Added super-basic support of `basepath` option.

There are two ways to setup a basepath:
- Global: `<Router basepath="/app">`
- Single location: `useLocation({ basepath: '/app' })`

### Common example
```jsx
<Router basepath="/app">
  <Link to="/">Index</Link>
  <Link to="/about">About Us</Link>

  <Route path="/">Index</Route>
  <Route path="/about">About Us</Route>
</Router>
```

### CODE REVIEW

**Comment by molefrog** [comment]
@omgovich Thanks! How far are we from the size limit?

File: test/use-location.test.js [hunk 1]
```diff
@@ -45,6 +45,16 @@ describe("`value` first argument", () => {
...
+  it("returns a pathname without a basepath", () => {
+    const { result, unmount } = renderHook(() =>
+      useLocation({ basepath: "/app" })
+    );
+
+    act(() => history.pushState(0, 0, "/app/dashboard"));
+    expect(result.current[0]).toBe("/dashboard");
+    unmount();
+  });
 });
```
Comment (molefrog) [inline test/use-location.test.js]: I would add a few extra test cases to handle the edge cases:
1. When the **basepath** is `/app` and current path is `/app` → `/`
2. When the **basepath** is `/app` and current path is `/home` → `/home`

File: use-location.js [hunk 2]
```diff
@@ -11,9 +12,12 @@ export default () => {
...
     // that's why we store the last pathname in a ref.
-    const checkForUpdates = () =>
-      prevPath.current !== location.pathname &&
-      update((prevPath.current = location.pathname));
+    const checkForUpdates = () => {
+      const pathname = currentPathname(basepath);
+      return (
+        prevPath.current !== pathname && update((prevPath.current = pathname))
+      );
+    };
```
[the remark this answers is not in the document]
Comment (omgovich) [reply use-location.js]: You're right! Good idea

**Comment by omgovich** [comment]
@molefrog 
```
  index.js
  Package size limit has exceeded by 61 B
  Size limit: 1.2 KB
  Size:       1.26 KB
  
  use-location.js
  Package size limit has exceeded by 50 B
  Size limit: 288 B
  Size:       338 B
```

**Comment by omgovich** [comment]
@molefrog 
```
  index.js
  Package size limit has exceeded by 53 B
  Size limit: 1.2 KB
  Size:       1.25 KB with all dependencies, minified and gzipped
  
  use-location.js
  Package size limit has exceeded by 50 B
  Size limit: 288 B
  Size:       338 B with all dependencies, minified and gzipped
```

File: use-location.js [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import { useEffect, useRef, useState, useCallback } from "./react-deps.js";
 
-export default (options = {}) => {
-  const { basepath = "" } = options;
+export default ({ basepath = "" } = {}) => {
```
Comment (molefrog) [inline use-location.js]: Nice! Was just about to propose this.

**Comment by molefrog** [comment]
I'm going to work on the types and README for this feature.
